### PR TITLE
[Tokens]: Update universal with Material Base

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Resolves https://github.com/brave/leo/issues/682

## TODO
- [x] Change button text color in dark mode
- [x] Fix toggle dark mode
- [x] Fix radiobutton dark mode
- [x] Fix checkbox dark mode

### CODE REVIEW

**Comment by aguscruiz** [comment]
What if instead of Static I use something like "Default"? 
-----
Yes, Android and iOS go in Universal too now.
-----
Is there any way we can see where does primitive colors were used? I'm not concerned about themable, but primitives will break
-----
There's something else I added over the last days, and it's the "blurple" and "orange" colors, in case we ever need to use them as they are, even if the theme is dynamic.

**Comment by fallaciousreasoning** [comment]
@aguscruiz wrt to static/default I actually got it working
same deal with the primitive colors - I think things should just work - gotta say this actually works really well!

**Comment by fallaciousreasoning** [comment]
Okay, I think this might be ready for a first review :fearful: @petemill @AlanBreck 

## Things I'd like to do:
1. Get reference resolving working with Style Dictionary - this seems to make the export work pretty darn well but the format that the variables needs to be in is a bit odd (we need to introduce a top level object to make it work).

**Comment by aguscruiz** [comment]
@fallaciousreasoning just wanted to double check something (out of ignorance)

With this update, would we get the advantage of aliasing tokens we talked about?

The one where for example, `text/primary` would ve defined as `themable/neutral/70` (or whatever it s) instead of the actual pure hex code?

**Comment by fallaciousreasoning** [comment]
Yup, that's right @aguscruiz - I changed the way we're doing it slightly (with a new `aliases` field) so we don't break the Android/iOS tokens. 

The stuff in the `value` field is essentially the default `Static` value while `aliases` tells it what variable it should point to

**Comment by aguscruiz** [comment]
Yayyy thanks ⭐⭐⭐⭐⭐

File: src/components/button/button.svelte [hunk 1]
```diff
@@ -389,15 +386,15 @@
     }
   }
   .leoButton.isPlainFaint {
-    --foreground: var(--leo-color-black);
+    --foreground: var(black);
```
Comment (fallaciousreasoning) [inline src/components/button/button.svelte:389]:
```suggestion
    --foreground: black;
```

File: src/tokens/transformation/web/color.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+const TinyColor = require('@ctrl/tinycolor')
+
+module.exports = {
+  type: 'value',
+  matcher(token) {
+    return token.type === 'color'
+  },
+  transformer({ value, referencedVariable }) {
+    if (referencedVariable) {
+      const name = referencedVariable.substring(1).replaceAll('.', '-')
```
Comment (petemill) [inline src/tokens/transformation/web/color.js:10]: Small point: This is the same as what we do with skia for referenced variables too, right (ignore the first char, and replace `.` with `-`? Should the knowledge of how the references variables translate to names be shared?

**Review by petemill — APPROVED** [review]
Amazing to get this done - thanks for taking it on

**Comment by fallaciousreasoning** [comment]
Merging... I'm scared :fearful:

**Pull request merged**